### ISSUE TITLE
[BUG] Bug in the imputer class. Fit and transform ignore the parameter for y 

### ISSUE THREAD

**Issue by ujohn33** [sktime/sktime#5284]
The `fit` and `transform` functions of the `Imputer` class in sktime ignore the input parameter for `y`. Upon debugging, it was found that `y` is always `None` and cannot be changed.

**To Reproduce**
```python
from sktime.datasets import load_airline
from sktime.transformations.series.impute import Imputer
model_reg = YfromX(lgb.LGBMRegressor())
model_reg.fit(x_train, y_train)
x_train = load_airline()
y_train = load_airline()
transformer = Imputer(method="forecaster", forecaster=model_reg)
transformer.fit(y_train, y=x_train)
y_train_imputed = transformer.transform(y_train, y=x_train)
```

**Expected behavior**
The `fit` and `transform` functions of the `Imputer` class should accept the input parameter for `y` and use it as exogenous regressors for the LGBMRegressor YfromX forecaster model. 

**Additional context**
I am using the editable development version of sktime, and the issue persists in this version.

```
System:
MacOS 12.5.1

Python dependencies:
pandas 12.1
numpy 1.25.2
lightgbm 4.0.0
sklearn 1.3.0

```

</details>

</details>

**Comment by fkiraly** [comment]
Confirmed on current main, python 3.11, windows.
Simplified code without external deps:

```python
from sklearn.linear_model import LinearRegression

from sktime.datasets import load_airline
from sktime.transformations.series.impute import Imputer

X = load_airline()
y = load_airline()

model_reg = YfromX(LinearRegression())
model_reg.fit(X, y)
transformer = Imputer(method="forecaster", forecaster=model_reg)

transformer.fit(y_train, y=x_train)
y_train_imputed = transformer.transform(y_train, y=x_train)
```

**Comment by fkiraly** [comment]
PS, you can get the `bug` badge for the bug report and the correct diagnosis (PR to allcontributorsrc)

The problem is indeed that passing `y` was always turned off via the tags. The fix is adding a tag that ensures passing of `y` if the `method` is `"forecaster"`.

**Comment by fkiraly** [comment]
Fixed in #5287, if you want to test it (will ship with next release, so you need to test from the branch until then)
